### DISCUSSION TITLE
Flexsearch implementation from Hextra (with highlighting)

### DISCUSSION THREAD

**Discussion by jollydodo** [thuliteio/doks#1186] [Ideas]
Hey, 

I just stumbled across the Hextra template and noticed that they also use Flexsearch. The code is really well annotated and their implementation is really nice in that it  structures the search results per category and more importantly: highlights the search input In the result sections. This is the main thing that I'm now missing in the current Hugo template, because Doks only shows the general top text and not the section of that is found on the page.

example: 
<img width="516" alt="image" src="https://github.com/gethyas/doks/assets/30504008/40ea8dbf-32b3-4ca8-8b08-31f6d459febd">
Trying to search for 3D scanner shows pages where to context of '3d scanner' is not clear from the result. For the pricelist result you would for example rather see the specific section where '3d scanner' is mentioned on that page: 
<img width="888" alt="image" src="https://github.com/gethyas/doks/assets/30504008/a077183b-9651-4dbe-b2d3-c4acb81e4678">

Hextra doesn't seem to do this either (not all the time at least?) but searching for a general term like "hextra" does clearly highlight sections and shows that the result is part of a specific docs section or the about page / blog for example: 
<img width="552" alt="image" src="https://github.com/gethyas/doks/assets/30504008/b2a2cd6d-8c1d-48da-b353-126dd6452fb7">
This makes it easier to select the right result

Also, the current implementation is perhaps aimed too much towards the docsearch implementation, because even the flexsearch version shows "recent searches" which is not applicable to the current flex search implementation: 
<img width="557" alt="image" src="https://github.com/gethyas/doks/assets/30504008/5f5226a4-a085-492f-bd18-e5325ee5740d">

Don't get me wrong, The doks search implementation works really well! It's lightning fast and does what it is supposed to do. I just thought that it might perhaps be useful to look at the Hextra version and code for small improvements. 

Template: 
https://imfing.github.io/hextra/

flexsearch implementation: 
https://github.com/imfing/hextra/blob/main/assets/js/flexsearch.js

Hextra uses Tainwindcss so we unfortunately can't use everything.

**Comment by h-enk** [comment]
That's a super nice theme!

It surely is a good idea to look at how Hextra did things — not only FlexSearch
